pkg/network/network_test.go

```diff
@@ -0,0 +1,128 @@
+package network
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"fmt"
+
+	osv1 "github.com/openshift/api/operator/v1"
+	v1 "k8s.io/api/core/v1"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+var _ = Describe("Testing network", func() {
+	// There is no functionality in this function yet
+	Describe("Canonicalize", func() {
+		Context("when given an empty config", func() {
+			conf := &opv1alpha1.NetworkAddonsConfigSpec{}
+			It("should pass", func() {
+				Canonicalize(conf)
+			})
+		})
+	})
+
+	Describe("Validate", func() {
+		Context("when given a valid config", func() {
+			conf := &opv1alpha1.NetworkAddonsConfigSpec{}
+			openshiftNetworkConf := &osv1.Network{}
+			It("should pass", func() {
+				err := Validate(conf, openshiftNetworkConf)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		Context("when given invalid config", func() {
+			conf := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{
+					RangeStart: "foo",
+				},
+				ImagePullPolicy: v1.PullPolicy("bar"),
+			}
+			openshiftNetworkConf := &osv1.Network{}
+			It("should return a compilation of errors", func() {
+				err := Validate(conf, openshiftNetworkConf)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("invalid configuration"))
+				Expect(err.Error()).To(ContainSubstring("both or none of the KubeMacPool ranges needs to be configured"))
+				Expect(err.Error()).To(ContainSubstring("requested imagePullPolicy 'bar' is not valid"))
+			})
+		})
+	})
+
+	Describe("FillDefaults", func() {
```

Comment:
I probably see why you don't have a negative case here (because the only way to receive an error from the function is if rand.Read() fails, which is hard to simulate); but in case you care to cover all branches (which seems like you effectively do with this PR), you could introduce a test mock / helper function for rand.Read that you could replace for test scenarios to trigger an error.

Reply:
I will just add a TODO, not that I am lazy, but I want to leave a low-hanging fruit for others (or future me)